File: LeetCode/455. Assign Cookies/Solution.java
import java.util.Arrays;

public class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int gLength = g.length, sLength = s.length, i = 0, j = 0, count = 0;
        while (i < gLength && j < sLength) {
            if (g[i] <= s[j]) {
                i++;
                j++;
                count++;
            } else {
                j++;
            }
        }
        return count;
    }
}

File: LeetCode/455. Assign Cookies/Solution.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        g.sort()
        s.sort()
        i = j = count = 0
        gLength, sLength = len(g), len(s)
        while i < gLength and j < sLength:
            if g[i] <= s[j]:
                i, j, count = i + 1, j + 1, count + 1
            else:
                j += 1
        return count
